test/common/router/router_ratelimit_test.cc

```diff
@@ -366,6 +366,94 @@ TEST_F(RateLimitPolicyEntryTest, RequestHeaders) {
               testing::ContainerEq(descriptors_));
 }
 
+// Validate that a descriptor is added if at least one request header has value.
+TEST_F(RateLimitPolicyEntryTest, RequestHeadersPartialMatchWithSkipIfAbsent) {
+  const std::string yaml = R"EOF(
+actions:
+- request_headers:
+    header_name: x-header-name
+    descriptor_key: my_header_name
+    skip_if_absent: false
+- request_headers:
+    header_name: x-header
+    descriptor_key: my_header
+    skip_if_absent: true
+  )EOF";
+
+  setupTest(yaml);
+  Http::TestHeaderMapImpl header{{"x-header-name", "test_value"}};
+
+  rate_limit_entry_->populateDescriptors(route_, descriptors_, "service_cluster", header,
+                                         default_remote_address_);
+  EXPECT_THAT(std::vector<Envoy::RateLimit::Descriptor>({{{{"my_header_name", "test_value"}}}}),
+              testing::ContainerEq(descriptors_));
+}
+
+TEST_F(RateLimitPolicyEntryTest, RequestHeadersPartialMatchWithoutSkipIfAbsent) {
```

Comment:
/s/RequestHeadersPartialMatchWithoutSkipIfAbsent/RequestHeadersWithDefaultSkipIfAbsent and add a comment above the test